codejail/jail_code.py

```diff
@@ -391,9 +391,17 @@ def save_artifacts(artifacts, save_path):
             path = images_dest_dir
         elif artifact_path.endswith(('.mp4', '.mov', '.gif', '.avi', '.mpeg', '.mkv')):
             path = videos_dest_dir
-        with open(artifact_path, 'r') as file:
-            content = file.read()
-            new_file = os.path.join(path, os.path.basename(artifact_path))
-        with open(new_file, 'w') as file:
-            file.write(content)
-        os.chmod(new_file, 0o777)
+
+        if artifact_path.endswith('.zip'):
+            import zipfile
```

Comment:
let's adhere to PEP8 - https://peps.python.org/pep-0008/#imports with imports at the top of the file